File: 2161_partition_array_according_to_given_pivot.java
class Solution {
    public int[] pivotArray(int[] x, int y) {
        List<Integer> a = new ArrayList<>(), b = new ArrayList<>(), c = new ArrayList<>();
        for (int z : x) (z < y ? a : z > y ? c : b).add(z);
        a.addAll(b); a.addAll(c);
        return a.stream().mapToInt(i -> i).toArray();
    }
}

File: 2161_partition_array_according_to_given_pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& x, int y) {
        vector<int> a, b, c;
        for (int z : x) (z < y ? a : z > y ? c : b).push_back(z);
        a.insert(a.end(), b.begin(), b.end());
        a.insert(a.end(), c.begin(), c.end());
        return a;
    }
};
